jaeger-spark-dependencies-test/src/main/java/io/jaegertracing/spark/dependencies/test/DependenciesTest.java

```diff
@@ -0,0 +1,161 @@
+import io.jaegertracing.spark.dependencies.test.tree.TreeGenerator;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.junit.Test;
+
+/**
+ * @author Pavol Loffay
+ */
+public abstract class DependenciesTest {
+
+  protected OkHttpClient okHttpClient = new OkHttpClient.Builder().build();
+  protected ObjectMapper objectMapper = JsonHelper.configure(new ObjectMapper());
+
+  /**
+   * Set these in subclasses
+   */
+  protected static String queryUrl;
+  protected static String collectorUrl;
+  protected static String zipkinCollectorUrl;
+
+  /**
+   * Override this and run spark job
+   */
+  protected abstract void deriveDependencies() throws Exception;
+
+  /**
+   * Wait between submitting different traces
+   */
+  protected abstract void waitBetweenTraces() throws InterruptedException;
+
+  @Test
+  public void testJaegerOneTrace() throws Exception {
+    TreeGenerator<Tracer> treeGenerator = new TreeGenerator(
+        TracersGenerator.generateJaeger(5, collectorUrl));
+    Node<JaegerWrapper> root = treeGenerator.generateTree(50, 3);
+    Traversals.inorder(root, (node, parent) -> node.getTracingWrapper().get().getSpan().finish());
+    waitBetweenTraces();
+    treeGenerator.getTracers().forEach(tracer -> {
+      tracer.getTracer().close();
+    });
+    waitJaegerQueryContains(root.getServiceName(), root.getTracingWrapper().operationName());
+
+    deriveDependencies();
+    assertDependencies(DependencyLinkDerivator.serviceDependencies(root));
+  }
+
+  @Test
+  public void testJaegerMultipleTraces() throws Exception {
+    TreeGenerator<Tracer> treeGenerator = new TreeGenerator(
+        TracersGenerator.generateJaeger(50, collectorUrl));
+    Map<String, Map<String, Long>> expectedDependencies = new LinkedHashMap<>();
+    for (int i = 0; i < 20; i++) {
+      Node<JaegerWrapper> root = treeGenerator.generateTree(50, 15);
+      DependencyLinkDerivator.serviceDependencies(root, expectedDependencies);
+      Traversals.inorder(root, (node, parent) -> node.getTracingWrapper().get().getSpan().finish());
+      waitBetweenTraces();
+      waitJaegerQueryContains(root.getServiceName(), root.getTracingWrapper().operationName());
+    }
+    // flush and wait for reported data
+    treeGenerator.getTracers().forEach(tracer -> tracer.getTracer().close());
+
+    deriveDependencies();
+    assertDependencies(expectedDependencies);
+  }
+
+//  @Test
+  public void testZipkinOneTrace() throws Exception {
+    TreeGenerator<Tracing> treeGenerator = new TreeGenerator(TracersGenerator.generateZipkin(5, zipkinCollectorUrl));
+    Node<ZipkinWrapper> root = treeGenerator.generateTree(150, 3);
+    Traversals.inorder(root, (node, parent) -> node.getTracingWrapper().get().getSpan().finish());
+    waitBetweenTraces();
+    treeGenerator.getTracers().forEach(tracer -> {
```

Comment:
Could be moved onto the `TreeGenerator`

Reply:
jaeger tracer does not implement `closeable` I will create a PR and add TODO here.